components/script/task_source/media_element.rs

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use dom::bindings::inheritance::Castable;
+use dom::bindings::refcounted::Trusted;
+use dom::event::{SimpleEventTask};
+use dom::eventtarget::EventTarget;
+use dom::window::Window;
+use msg::constellation_msg::PipelineId;
+use script_runtime::{CommonScriptMsg, ScriptThreadEventCategory};
+use script_thread::MainThreadScriptMsg;
+use servo_atoms::Atom;
+use std::fmt;
+use std::result::Result;
+use std::sync::mpsc::Sender;
+use task::{TaskCanceller, TaskOnce};
+use task_source::{TaskSource, TaskSourceName};
+
+#[derive(Clone, JSTraceable)]
+pub struct MediaElementTaskSource(pub Sender<MainThreadScriptMsg>, pub PipelineId);
```

Comment:
This `Sender` needs to be a `servo_channel::Sender` instead of a `std::sync::mpsc::Sender`